roles/zuul/files/jobs/bonnyci.yaml

```diff
@@ -0,0 +1,43 @@
+- publisher:
+    name: bonnyci-logs
+    publishers:
+      - scp:
+          site: bonnyci-scp
+          files:
+            - target: 'logs/$LOG_PATH'
+              source: '/opt/bonnyci/logs/**'
+              keep-hierarchy: true
+              copy-after-failure: true
+
+- job-template:
+    name: 'bonnyci-run-{pipeline}'
+    pipeline: '{pipeline}'
+    test_log_dir: /opt/bonnyci/logs
+    builders:
+      - zuul-git-prep
+      - shell: |
+            #!/bin/bash
+            if [[ ! -f .bonnyci/run.sh ]]; then
+                echo "ERROR: No test script found at .bonnyci/run.sh, skipping"
+                exit 1
+            elif [[ ! -x .bonnyci/run.sh ]]; then
+                echo "ERROR: Test script found at .bonnyci/run.sh but it is not executable"
+                exit 1
+            fi
+
+            export BONNYCI_TEST_PIPELINE={pipeline}
+            export BONNYCI_TEST_LOG_DIR={test_log_dir}
+
+            (cd .bonnyci/ && ./run.sh)
```

Comment:
I think i would expect to be in the root directory of the code checkout. That root directory is not as predeictable as we might like, it's something like ~/workspace/{pipeline}/{repo} so it's not easy to go back to that dir to run something like tox. 

On the other hand if you make .bonnyci a well known folder location it's super simple for people in scripts to do source .bonnyci/abc.sh. 

I would vote for remaining in the checkout dir.